server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1463,19 +1463,26 @@ public final void parse(String name, MappingParserContext parserContext, Map<Str
                 }
             }
             String type = (String) fieldNode.remove("type");
-            for (Iterator<Map.Entry<String, Object>> iterator = fieldNode.entrySet().iterator(); iterator.hasNext();) {
+
+            List<String> order = Arrays.stream(params).map(p -> p.name).toList();
+            Iterator<Map.Entry<String, Object>> iterator = fieldNode.entrySet().stream().sorted((e1, e2) -> {
+                int i1 = order.indexOf(e1.getKey());
+                int i2 = order.indexOf(e2.getKey());
+                return Integer.compare(i1, i2);
+            }).iterator();
+            while (iterator.hasNext()) {
```

Comment:
I am against this, let's remove to try and find a better fix if we still run into weird parsing ordering failures. 

It would be better to update the dense mapper validations than to fundamentally change how all mappers iterate values.

Reply:
OK, I understand your concerns. I will move the commits related to the mapper bug to a separate PR for further discussion. This PR will remain focused solely on fixing the Synthetic Source issue. Does that sound good to you?

Reply:
OK, new PR https://github.com/elastic/elasticsearch/pull/114516